gammapy/maps/wcs.py

```diff
@@ -882,6 +882,20 @@ def __repr__(self):
         )
         return str_
 
+    def __eq__(self, other):
+        """Test equality between two `~gammapy.maps.WcsGeom`. Espect tolerance of 1e-6 for `~astropy.wcs.WCS` objects"""
+        if not isinstance(other, self.__class__):
+            raise TypeError('Cannot compare WcsGeom with {}'.format(other.__class__))
+
+        # check overall shape and axes compatibility
+        if self.data_shape != other.data_shape:
+            return False
+
+        result = True
```

Comment:
Also here: I would not introduce a local state variable "result" that you accumulate. Just check things and return False if not equal one by one, and put `return True` on the last line.
I think I've seen that coding pattern a lot in Python codes.

Reply:
Done